sdk/iothub/src/service/requests/create_or_update_module_identity_builder.rs

```diff
@@ -0,0 +1,193 @@
+use std::convert::TryInto;
+
+use azure_core::errors::AzureError;
+
+use http::Method;
+use serde::Serialize;
+
+use crate::service::resources::{
+    identity::IdentityOperation, AuthenticationMechanism, AuthenticationType, SymmetricKey,
+    X509ThumbPrint,
+};
+
+use crate::service::responses::ModuleIdentityResponse;
+use crate::service::{IoTHubError, ServiceClient, API_VERSION};
+
+/// The CreateOrUpdateModuleIdentityBuilder is used to construct a new module identity
+/// or the update an existing one.
+pub struct CreateOrUpdateModuleIdentityBuilder<'a> {
+    service_client: &'a ServiceClient,
+    authentication: Option<AuthenticationMechanism>,
+    device_id: Option<String>,
```

Comment:
Nit: This is fine for now. We need to make a decision across crates whether these builders generally take ownership of the data or borrow them. This builder already has a lifetime associated with the client. We could conceivably have these `String`s be `&'a str` instead.